elements/lisk-chain/src/block.ts

```diff
@@ -76,5 +77,26 @@ export class Block {
 		for (const tx of this.payload) {
 			tx.validate();
 		}
+
```

Comment:
I think assetroot and tx root needed to be validated  here.
However, to do this in sync func, we need to create function to calculate merkle root. Ill create  separate  issue for this

Reply:
for trx root, if we do it here then we can remove its validation from the generator I think

Reply:
Will be resolved in the issue you created https://github.com/LiskHQ/lisk-sdk/issues/6886